domain/secretbackend/params.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package secretbackend
+
+import (
+	"time"
+
+	"github.com/juju/juju/core/secrets"
+)
+
+// UpsertSecretBackendParams are used to upsert a secret backend.
+type UpsertSecretBackendParams struct {
+	ID                  string
+	Name                string
+	BackendType         string
+	TokenRotateInterval *time.Duration
+	NextRotateTime      *time.Time
+	Config              map[string]interface{}
+}
+
+// CreateSecretBackendParams are used to create a secret backend.
+type CreateSecretBackendParams struct {
+	ID                  string
+	Name                string
+	BackendType         string
+	TokenRotateInterval *time.Duration
+	NextRotateTime      *time.Time
+	Config              map[string]interface{}
+}
+
+// UpdateSecretBackendParams are used to update a secret backend.
+type UpdateSecretBackendParams struct {
```

Comment:
Is this used now we have UpsertParams?
Same with Create?

Reply:
It'd be better to have `CreateSecretBackend` and `UpdateSecretBackend` in the service layer and they all call `state.UpsertSecretBackend`.
If you see the difference between `CreateSecretBackendParams ` and `UpdateSecretBackendParams`, you will know the benefit.

Reply:
We want to be careful not to mix service params and state params. Given that this PR is for the state layer, we shouldn't be adding service structs that are unused in this PR. We are still to decide where service params structs should go - some places have them in the service package. If they are to live in this top level domain package, they should at least be in a separate file otherwise it's very confusing for the reader to understand what's used for a service caller vs service calling to state. But since these are currently unused in this PR, they should be removed.